mllib/src/main/scala/org/apache/spark/ml/Evaluator.scala

```diff
@@ -36,5 +35,18 @@ abstract class Evaluator extends Identifiable {
    * @param paramMap parameter map that specifies the input columns and output metrics
    * @return metric
    */
-  def evaluate(dataset: DataFrame, paramMap: ParamMap): Double
+  def evaluate(dataset: DataFrame, paramMap: ParamMap): Double = {
+    this.copy(paramMap).evaluate(dataset)
+  }
+
```

Comment:
`Evaluator` is used for tuning model, where we need a scalar metric to select the best model. We can try returning a comparable `T`, but this is not part of this PR.